server/semantics-service/src/main/java/edu/illinois/ncsa/incore/service/semantics/controllers/TypeController.java

```diff
@@ -47,44 +47,55 @@ public class TypeController {
 
     private final String username;
 
-    private final Authorizer authorizer;
     private final List<String> groups;
 
+    private String endpointUrl = "https://incore.ncsa.illinois.edu/semantics/api/types/";
+
     @Inject
     private ITypeDAO typeDAO;
 
     @Inject
     private ISpaceRepository spaceRepository;
 
+    @Inject
+    private IAuthorizer authorizer;
+
     @Inject
     public TypeController(
         @Parameter(name = "User credentials.", required = true) @HeaderParam("x-auth-userinfo") String userInfo,
         @Parameter(name = "User groups.", required = false) @HeaderParam("x-auth-usergroup") String userGroups
     ) {
         this.username = UserInfoUtils.getUsername(userInfo);
         this.groups = UserGroupUtils.getUserGroups(userGroups);
-        // we want to limit the semantics service to admins for now
-        this.authorizer = new Authorizer();
-        if (!this.authorizer.isUserAdmin(this.groups)) {
-            throw new IncoreHTTPException(Response.Status.FORBIDDEN, this.username + " is not an admin.");
-        }
     }
 
     @GET
     @Path("types")
     @Produces({MediaType.APPLICATION_JSON})
     @Operation(summary = "list all types belong user has access to.")
-    public Response listTypes() {
+    public Response listTypes(
+        @Parameter(name = "Specify the order of sorting, either ascending or descending.") @DefaultValue("asc") @QueryParam("order") String order,
+        @Parameter(name = "Skip the first n results.") @DefaultValue("0") @QueryParam("skip") int offset,
+        @Parameter(name = "Limit number of results to return.") @DefaultValue("50") @QueryParam("limit") int limit,
+        @Parameter(name = "List the details.") @DefaultValue("false") @QueryParam("detail") boolean detail) {
```

Comment:
could we use hyperlink instead of detail? I got confused as I thought detail meant to get the entire details of each document. Since we return just a list of titles when detail=false, we can say hyperlink=false/true which I think can make it clearer to other users of the API